src/main/java/com/gregtechceu/gtceu/integration/ae2/machine/MEPatternBufferProxy.java

```diff
@@ -0,0 +1,195 @@
+package com.gregtechceu.gtceu.integration.ae2.machine;
+
+import com.gregtechceu.gtceu.GTCEu;
+import com.gregtechceu.gtceu.api.GTValues;
+import com.gregtechceu.gtceu.api.capability.recipe.IO;
+import com.gregtechceu.gtceu.api.machine.IMachineBlockEntity;
+import com.gregtechceu.gtceu.api.machine.MetaMachine;
+import com.gregtechceu.gtceu.api.machine.feature.IMachineLife;
+import com.gregtechceu.gtceu.api.machine.multiblock.part.TieredIOPartMachine;
+import com.gregtechceu.gtceu.api.machine.trait.*;
+
+import com.lowdragmc.lowdraglib.gui.editor.runtime.PersistedParser;
+import com.lowdragmc.lowdraglib.gui.modular.ModularUI;
+import com.lowdragmc.lowdraglib.syncdata.annotation.Persisted;
+import com.lowdragmc.lowdraglib.syncdata.annotation.ReadOnlyManaged;
+import com.lowdragmc.lowdraglib.syncdata.field.ManagedFieldHolder;
+
+import net.minecraft.core.BlockPos;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.phys.BlockHitResult;
+
+import lombok.Getter;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+import static com.gregtechceu.gtceu.integration.ae2.machine.MEPatternBufferPartMachine.MAX_PATTERN_COUNT;
+
+@SuppressWarnings("unused")
+public class MEPatternBufferProxy extends TieredIOPartMachine implements IMachineLife {
+
+    protected static final ManagedFieldHolder MANAGED_FIELD_HOLDER = new ManagedFieldHolder(
+            MEPatternBufferProxy.class, TieredIOPartMachine.MANAGED_FIELD_HOLDER);
+
+    @Getter
+    @ReadOnlyManaged(onDirtyMethod = "onDirty", serializeMethod = "onSave", deserializeMethod = "onLoadCircuit")
+    protected NotifiableItemStackHandler circuitInventorySimulated;
+
+    @Getter
+    @ReadOnlyManaged(onDirtyMethod = "onDirty", serializeMethod = "onSave", deserializeMethod = "onLoadShare")
+    protected NotifiableItemStackHandler shareInventory;
+
+    @Getter
+    @ReadOnlyManaged(onDirtyMethod = "onDirtyFluid", serializeMethod = "onSaveFluid", deserializeMethod = "onLoadFluid")
+    protected NotifiableFluidTank shareTank;
+
+    @Getter
+    @ReadOnlyManaged(onDirtyMethod = "onDirtyInternal",
+                     serializeMethod = "onSaveInternal",
+                     deserializeMethod = "onLoadInternal")
+    protected MEPatternBufferPartMachine.InternalSlot[] internalInventory = new MEPatternBufferPartMachine.InternalSlot[MAX_PATTERN_COUNT];
```

Comment:
remove the initialization, it goes unused anyway and can break the saving logic.